src/state/slices/limitOrderSlice/limitOrderSlice.ts

```diff
@@ -0,0 +1,211 @@
+import type { PayloadAction } from '@reduxjs/toolkit'
+import { createSlice } from '@reduxjs/toolkit'
+import type { QuoteId } from '@shapeshiftoss/types/dist/cowSwap'
+import assert from 'assert'
+import type { InterpolationOptions } from 'node-polyglot'
+
+import { TransactionExecutionState } from '../tradeQuoteSlice/types'
+import {
+  initialState,
+  limitOrderSubmissionInitialState,
+  LimitOrderSubmissionState,
+} from './constants'
+import { buildUnsignedLimitOrder } from './helpers'
+import type { LimitOrderActiveQuote } from './types'
+
+export const limitOrderSlice = createSlice({
+  name: 'limitOrder',
+  initialState,
+  reducers: {
+    clear: state => ({
+      ...initialState,
+      orderSubmission: state.orderSubmission, // Leave the limit order submission state alone
+    }),
+    setActiveQuote: (state, action: PayloadAction<LimitOrderActiveQuote | undefined>) => {
+      if (action.payload === undefined) {
+        state.activeQuote = undefined
+        return
+      }
+      const quoteId = action.payload.response.id
+
+      if (!quoteId) {
+        console.error('Missing quoteId')
+        return
+      }
+      state.activeQuote = action.payload
+      state.orderSubmission[quoteId] = limitOrderSubmissionInitialState
+    },
+    setLimitOrderInitialized: (state, action: PayloadAction<QuoteId>) => {
+      state.orderSubmission[action.payload].state = LimitOrderSubmissionState.Previewing
+    },
+    confirmSubmit: (state, action: PayloadAction<QuoteId>) => {
+      const limitOrderQuoteId = action.payload
+
+      // If there is no active quote, the state is corrupt.
+      if (state.activeQuote === undefined) {
+        console.error('Attempted to confirm an non-existent limit order quote')
+        return
+      }
+
+      // If this is not true, the state is corrupt.
+      assert(state.activeQuote.response.id === limitOrderQuoteId)
+
+      const unsignedOrderCreation = buildUnsignedLimitOrder(state.activeQuote)
+      state.confirmedLimitOrder[limitOrderQuoteId] = {
+        params: state.activeQuote.params,
+        unsignedOrderCreation,
+      }
+
+      if (state.orderSubmission[limitOrderQuoteId].state !== LimitOrderSubmissionState.Previewing) {
+        if (
+          state.orderSubmission[limitOrderQuoteId].state === LimitOrderSubmissionState.Initializing
+        ) {
+          console.error('Attempted to confirm an uninitialized limit order')
+        } else {
+          console.error('Attempted to confirm an in-progress limit order')
+        }
+        return
```

Comment:
The intent here is really just to softly fail with console errors rather than crashing. In my mind if there is an issue, we get some debugging hint without totally exploding but I'm open to trying a different approach? This is the approach we're using for spot trades also.

Reply:
@woodenfurniture pinging you there to make sure we don't miss anything, resolving anyway as It doesn't looks like a blocker